runtime/event_loop.cpp

```diff
@@ -84,7 +84,7 @@ bool EventLoop::run_event_loop(api::Engine *engine, double total_compute) {
     }
 
     // Select the next task to run according to event-loop semantics of oldest-first.
-    size_t task_idx = api::AsyncTask::select(tasks);
+    size_t task_idx = api::AsyncTask::select(*tasks);
```

Comment:
I really like that the assumption about `tasks` being non-null is pushed out to here, rather than being something that `select` might need to care about :+1: